sdks/go/pkg/beam/core/runtime/exec/sdf.go

```diff
@@ -678,10 +678,6 @@ func (n *ProcessSizedElementsAndRestrictions) Checkpoint() ([]*FullValue, error)
 		return nil, addContext(err)
 	}
 
-	if !n.rt.IsDone() {
-		return nil, addContext(errors.Errorf("Primary restriction %#v is not done. Check that the RTracker's TrySplit() at fraction 0.0 returns a completed primary restriction", n.rt))
-	}
```

Comment:
I had to remove this check because the new splitting logic ends up setting `rest.End = rest.claimed + 1`. Since the claimed work _should_ be complete, this means that the restriction is done, but that condition can't be checked here without adding new methods for users to implement (which doesn't seem worth it).

Reply:
I don't love removing this check since I think it has value in preventing a misconfigured RTracker from losing data. I also think that new behavior works with this if the `IsDone()` function for the offsetrange tracker is tweaked like I suggested

Reply:
Glad this was caught. This is basically the only place we can tell the user if they've implemented the SDF incorrectly in some fashion. While it's later than ideal (at execution, rather than construction time), a hard abort is preferable to other correctness or dataloss issues.